src/components/app/ServiceWorkerManager.js

```diff
@@ -126,26 +126,26 @@ class ServiceWorkerManagerImpl extends PureComponent<Props, State> {
       });
     });
 
-    wb.addEventListener('activated', () => {
+    wb.addEventListener('controlling', () => {
       // The update could have been triggered by this tab or another tab.
       // We distinguish these cases by looking at this.state.installStatus.
       console.log(
-        '[ServiceWorker] The new version of the application has been enabled.'
+        '[ServiceWorker] The new version of the application has started handling the fetch requests.'
       );
 
       if (this.state.installStatus === 'activating') {
-        // In this page the user clicked on the "reload" button.
+        // In this page the user clicked on the "Apply and reload" button.
         this.reloadPage();
         return;
       }
 
-      // In another page, the user clicked on the "reload" button.
+      // In another page, the user clicked on the "Apply and reload" button.
 
       const ready =
         !this._hasDataSourceProfile() || this._isProfileLoadedAndReady();
 
       this.setState({
-        installStatus: 'activated',
+        installStatus: 'controlling',
         // But if we weren't quite ready, we should write it in the notice.
         updatedWhileNotReady: !ready,
```

Comment:
At first I wasn't 100% sure about either keeping this case inside the 'activated' event handler or moving to `controlling`. But then I've realized that this refresh bug also affects this case.
For example when you have another profiler.firefox.com tab, and got a new version. When you refresh it without clicking the "apply and reload" button, this case also starts to not work since it doesn't dispatch the `activated` event at all.